src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java

```diff
@@ -963,11 +964,14 @@ public List<ControlledVocabularyValue> parseControlledVocabularyValue(DatasetFie
                 }
                 String strValue = json.getString("value", "");
                 ControlledVocabularyValue cvv = datasetFieldSvc.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(cvvType, strValue, lenient);
-                if (cvv == null) {
+                if (cvv == null && !allowHarvestingMissingCVV) {
                     throw new ControlledVocabularyException("Value '" + strValue + "' does not exist in type '" + cvvType.getName() + "'", cvvType, strValue);
                 }
-                return Collections.singletonList(cvv);
+                if (cvv != null) {
+                    vals.add(cvv);
+                }
```

Comment:
It's clear that the changes above allow a cvv==null to propagate instead of throwing an exception and the if clause here just skips adding a null value to vals. So this should just ignore non-CVV values w/o blowing up.

I think the intended scope would be to also keep the non-CVV val for indexing which would be ~like an else clause here to add a DatasetFieldValue (as in https://github.com/IQSS/dataverse/blob/f456e515e90e15eed1f1ef6f089e6b764c61df66/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java#L876-L877) which would allow you to keep the value so it could be indexed (without adding a new CVV value to the defined ones which has the side effect you note of adding this term as a new CVV value that can be added in non-harvested datasets.) 

An else clause here wouldn't work as DatasetFieldValue and ControlledVocabularyValue don't have a common subclass (and aren't saved in the same array in the DatasetField class). What might work instead, would be to just call the parsePrimitiveValue method instead of this one when it is a Harvested dataset and allowHarvestingMissingCVV is true. This would make the DatasetField have only DatasetFieldValues, even for the ones where a CVV did exist. The hope is that since these fields are only used for indexing in harvested datasets, there would not be any side effects of having DatasetFieldValues instead of CVVs in this case. I don't think that's proven yet, but I think this is something you could/should try. If it gets messy (i.e. if we make assumptions in the indexing code about whether a DatasetField for a CVV DatasetFieldType has to have CVV values and you can't easily add the same allowHarvestingMIssingCVV flag there), it may be worth reconsidering the scope/priority.

Reply:
restructured the code to keep the non CVVs in the DatasetFieldValues list instead of skipping them so that the can be indexed later